src/pages/billing/SpendReport/SpendReport.test.ts

```diff
@@ -41,55 +45,65 @@ describe('SpendReport', () => {
   const otherCostMessaging = /other infrastructure or query costs related to the general operations of Terra/i
 
   const createSpendReportResult = totalCost => {
-    return {
-      spendSummary: {
-        cost: totalCost, credits: '2.50', currency: 'USD', endTime: 'dummyTime', startTime: 'dummyTime'
-      },
-      spendDetails: [
+    const categorySpendData = {
+      aggregationKey: 'Category',
+      spendData: [
+        { cost: '999', category: 'Compute' },
```

Comment:
@blakery I decided to make sure that my mock data was meeting the TS definitions, which actually pointed out that I had a mistake in the TS definition related to endTime/startTime (which is never used).

But for some reason it isn't enforcing things in the way I would expect. For instance, it doesn't complain here that `currency` and `credits` are missing (again, they aren't actually used). And what bothers me more is that after I [added the specific possible values](https://github.com/DataBiosphere/terra-ui/pull/3842/commits/cb38761f8d6132c791abd708d1d34568ce9a1c5c#diff-9e5e64e26922f85475ed7d4fe6939200cd82200da840b5d6aa1cb85aa3167d18R51) for `category`, it doesn't complain if I change `category` to some bogus value. It also doesn't complain if I add undeclared properties other to the spendData.

Do you know what is going on? I'm also going to post in terra-ui to see if folks have ideas.

Reply:
@blakery never mind, it's because I was using "as …" instead of declaring the type on the variable declaration. https://broadinstitute.slack.com/archives/C01EHNUM73R/p1678456694932509